version: "3.4"
   
services:
  database:
   image: kartoza/postgis:9.6-2.4
   volumes:
      - ./data/db:/var/lib/postgresql/data
   environment:
      POSTGRES_USER: $${DATABASE_USER_PROD}
      POSTGRES_PASS: $${DATABASE_PASS_PROD}
      POSTGRES_DB: $${DATABASE_NAME_PROD}
      
   ports:
   - "5432:5432"


  backend:
     build: 
      context: backend
      dockerfile: Dockerfile
     container_name: backend
     command: bash -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
     environment:
       DATABASE_HOST: database
       DATABASE_USER: $${DATABASE_USER_PROD}
       DATABASE_PASS_PROD: $${DATABASE_PASS_PROD}
       DATABASE_NAME: $${DATABASE_NAME_PROD}
       AWS_ACCES_KEY_ID: $${AWS_ACCES_KEY_ID}
       AWS_SECRET_ACCES_KEY: $${AWS_SECRET_ACCES_KEY}
       SECRET_KEY: $${DJANGO_SECRET_KEY}
       ENVIRONMENT: $${ENVIRONMENT}
       MAIL_USER: $${MAIL_USER}
       MAIL_KEY: $${MAIL_KEY}
       EMAIL_REGISTER: $${EMAIL_REGISTER}

     volumes:
       - ./backend/app/:/src/app
     ports:
     - "8000:8000"
     restart: on-failure
     links:
       - database
